addons/website/migrations/12.0.1.0/post-migration.py

```diff
@@ -32,45 +32,12 @@ def fill_website_socials(cr):
     )
 
 
-def apply_bootstrap_4(view):
-    text = view.arch_db
-    # TO BE FILLED
-    view.arch_db = text
-
-
-def bootstrap_4_migration(env):
-    pages = env['website.page'].search([])
-    views = pages.mapped('view_id').filtered(
-        lambda v: v.type == 'qweb' and not v.xml_id)
-    for view in views:
-        apply_bootstrap_4(view)
-
-
-def apply_copy_views(env):
```

Comment:
I don't get why you delete this method. But if you are sure with this, then please update the `openupgrade_analysis_work.txt` file with your reasons.

Reply:
It was not complete (only website pages were handled, but not the rest of the website views).

Reply:
Yes, and the valid logic inside has been moved to pre stage.

For instance, the contactus page is marked as noupdate (so it shouldn't be touched by migration scripts), but [it's included in the noupdate records that should change](https://github.com/OCA/OpenUpgrade/blob/069f5623491fe9b90264b8e5776b724a833a4ec4/addons/website/migrations/12.0.1.0/noupdate_changes.xml#L12) (because of the inherent logic inside the view). So, if we copied+migrated views in the post stage, that one would be already modified and thus lost.

Instead of that, we duplicate and migrate to BS4 in the pre stage, and then the rest of normal migration happens as usual. If i.e. the contactus page for website 1 is broken, you just disable it and revert to its website-agnostic version, just as it came with the module.